src/main/java/org/opensearch/ad/AnomalyDetectorExtension.java

```diff
@@ -37,6 +111,457 @@ public ExtensionSettings getExtensionSettings() {
         return this.settings;
     }
 
+    private static Void initGson() {
+        gson = new GsonBuilder().serializeSpecialFloatingPointValues().create();
+        return null;
+    }
+
+    /**
+     * Returns components added by this extension.
+     *
+     * @param client A client to make requests to the system
+     * @param clusterService A service to allow watching and updating cluster state
+     * @param threadPool A service to allow retrieving an executor to run an async action
+     * @param environment the environment for path and setting configurations
+     */
+    public Collection<Object> createComponents(
+        SDKClient client,
+        ClusterService clusterService,
+        ThreadPool threadPool
+    ) {
+        EnabledSetting.getInstance().init(clusterService);
+        /* @anomaly-detection.create-detector
+        NumericSetting.getInstance().init(clusterService);
+        this.client = client;
+        this.threadPool = threadPool;
+        */
+        TransportService transportService = extensionsRunner.extensionTransportService;
+        Settings settings = extensionsRunner.sendEnvironmentSettingsRequest(transportService);
+        /* @anomaly-detection.create-detector
+        Throttler throttler = new Throttler(getClock());
+        this.clientUtil = new ClientUtil(settings, client, throttler);
+        this.indexUtils = new IndexUtils(client, clientUtil, clusterService, indexNameExpressionResolver);
+        this.nodeFilter = new DiscoveryNodeFilterer(clusterService);
+        */
+        // AnomalyDetectionIndices is Injected for IndexAnomalyDetectorTrasnportAction constructor
+        this.anomalyDetectionIndices = new AnomalyDetectionIndices(
+            client,
```

Comment:
Rather than passing the `SDKClient` directly to component constructors, we'll first have to initialize an `OpenSearchClient` using [`SDKClient::initializeClient`](https://github.com/opensearch-project/opensearch-sdk-java/blob/5ecbd7121abbcf8a2c126d588d7771a86199baa2/src/main/java/org/opensearch/sdk/SDKClient.java#L35) and pass this object instead.